📝 Docs : basecocktail 요구사항정의서 초안 및 용어사전 초안 작성

YoungH66 opened the pull request:
## 📝 변경 사항
<!---- 변경 사항 및 관련 이슈에 대해 간단하게 작성해주세요. ---->
- basecocktail 요구사항 정의서와 용어사전의 초안입니다.
## #️⃣연관된 이슈
<!---- Ref or Resolves : #(Issues Number) ---->

### 스크린샷 (선택)
<!---- 첨부할 이미지가 있다면 추가해주세요. ---->

## 💬리뷰 요구사항(선택)
<!---- 리뷰어가 확인해 주었으면 하는 부분이 있다면 작성해주세요. ---->
- 디자인 시안에 따르면, basecocktail(디자인 시안에선 original)과 customcocktail이 동시에 보여지고 있어서, 조율이 필요할 듯 합니다.
- 찜하기 기능은 basecocktail보다 user에서 담당하는게 맞는거 같지만, 차후 정렬등 기능에 사용할 수 있을듯 싶어, 언급해두고 있습니다.
- 상세정보 - 칵테일 정보 - 탄산 여부는 재료를 참고해서 표시 할 수 도 있지만, 일관적이지 못한것 같습니다.
- 칼로리의 경우 고려하지 못한 요소이기 때문에 의논이 필요합니다

## 🔎 PR 유형
- [ ] 새로운 기능 추가
- [ ] 버그 수정
- [ ] CSS 등 사용자 UI 디자인 변경
- [ ] 코드에 영향을 주지 않는 변경사항
- [ ] 코드 리팩토링
- [ ] 주석 추가 및 수정
- [x] 문서 수정
- [ ] 테스트 추가, 테스트 리팩토링
- [ ] 빌드 부분 혹은 패키지 매니저 수정
- [ ] 파일 혹은 폴더명 수정
- [ ] 파일 혹은 폴더 삭제

YoungH66 committed:
📝 Docs : basecocktail 요구사항정의서 초안 및 용어사전 초안 작성

basecocktail 요구사항 정의서와 용어사전의 초안입니다.

yeossan commented:
basecocktail, customcocktail이 동시에 보이는건 이야기를 해봐야할거같아요! 고생하셨습니다.

munkm1109 commented on base-cocktail-SRS.md:
```diff
@@ -0,0 +1,66 @@
+
+### 칵테일
+- `칵테일`을 등록할 수 있다.
+- `칵테일 이름`이 올바르지 않으면 등록할 수 없다.
```
선행으로, 올바르다에 대한 기준이 먼저 정의되어야 할 것 같습니다.
Ex) 올바르다는 기존에 존재하는 칵테일 이름과 일치함을 의미한다.

sinwoongkang1 reviewed (commented):
고생하셨습니다. 요구사항정의서 참고 많이 할게요!!

munkm1109 commented on base-cocktail-SRS.md:
```diff
@@ -0,0 +1,66 @@
+# 칵테일 관리 시스템 요구사항 정의서
+
+### 칵테일
+- `칵테일`을 등록할 수 있다.
```
여기서 "칵테일"이 의미하는 것이, 
"칵테일 이름"과 다른 점이 무엇인가요?

munkm1109 replied:
재료, 칵테일이름, 레시피 등 후술되는 모든 '기능으로써 작용하는 명사'에 대해서는 정의가 필요합니다.

munkm1109 commented on base-cocktail-SRS.md:
```diff
@@ -0,0 +1,66 @@
+    - `칵테일 이름`은 비워둘 수 없다.
+- `재료`가 올바르지 않으면 등록할 수 없다.
+    - 최소 1개 이상의 `재료`를 가진다.
+    - 각 `재료`는 `용량`을 가진다.
```
용량도 마찬가지로, 사용되는 단위를 모두 기입하는 등, 정의가 필요합니다.

munkm1109 replied:
용량의 경우, 사용되는 단위가 무엇인지 등의 조건이 "칵테일 이름"처럼 필요할 것 같습니다.

munkm1109 commented on base-cocktail-SRS.md:
```diff
@@ -0,0 +1,66 @@
+- `재료`가 올바르지 않으면 등록할 수 없다.
+    - 최소 1개 이상의 `재료`를 가진다.
+    - 각 `재료`는 `용량`을 가진다.
+- `레시피`가 올바르지 않으면 등록할 수 없다.
```
레시피 등록이, 저희가 기본 칵테일 레시피를 등록할 때의 시나리오인지, 고객이 커스텀 칵테일을 등록할 때의 시나리오인지 명확하지 않습니다.
커스텀 칵테일 레시피를 등록하는 시나리오로, 우선 작성해보는 것이 좋을 것 같습니다. (보류)

>>>>>>>>Basecoctail요구사항 정의서 범위가 어디까지인지? 여쭤보기

munkm1109 commented on base-cocktail-SRS.md:
```diff
@@ -0,0 +1,66 @@
+    - 최소 1개 이상의 `재료`를 가진다.
+    - 각 `재료`는 `용량`을 가진다.
+- `레시피`가 올바르지 않으면 등록할 수 없다.
+    - 최소 1단계 이상의 `제조 단계`를 가진다.
```
>>>>>제조단계란 무엇인지? 여쭤보고, 조율해서 작성하기

munkm1109 commented on base-cocktail-SRS.md:
```diff
@@ -0,0 +1,66 @@
+    - 각 `재료`는 `용량`을 가진다.
+- `레시피`가 올바르지 않으면 등록할 수 없다.
+    - 최소 1단계 이상의 `제조 단계`를 가진다.
+- `칵테일 정보`가 올바르지 않으면 등록할 수 없다.
```
칵테일 정보 / 칵테일 / 칵테일이름 간의 차이가 무엇인가요?

munkm1109 commented on base-cocktail-SRS.md:
```diff
@@ -0,0 +1,66 @@
+- `레시피`가 올바르지 않으면 등록할 수 없다.
+    - 최소 1단계 이상의 `제조 단계`를 가진다.
+- `칵테일 정보`가 올바르지 않으면 등록할 수 없다.
+    - `알코올 도수`를 가진다.
```
알코올 도수란, 원액기준인가? 혹은 계산된 도수가 기준인가?에 대한 정의가 필요합니다.

munkm1109 commented on base-cocktail-SRS.md:
```diff
@@ -0,0 +1,66 @@
+    - 최소 1단계 이상의 `제조 단계`를 가진다.
+- `칵테일 정보`가 올바르지 않으면 등록할 수 없다.
+    - `알코올 도수`를 가진다.
+    - `맛`을 가진다.
```
저희가 정했던 맛을 분류하기로 한 기준이 필요합니다.
Ex) 맛은 단맛, 짠맛, 신맛, 매운맛 으로 구성됩니다.
- 단맛은 상 중 하로 나뉘어집니다. 
등등

munkm1109 commented on base-cocktail-SRS.md:
```diff
@@ -0,0 +1,66 @@
+- `칵테일 정보`가 올바르지 않으면 등록할 수 없다.
+    - `알코올 도수`를 가진다.
+    - `맛`을 가진다.
+    - `분류`를 가진다.
```
분류란 무엇인지 모르겠습니다. 분류에 대한 정의가 필요합니다.

munkm1109 commented on base-cocktail-SRS.md:
```diff
@@ -0,0 +1,66 @@
+    - `맛`을 가진다.
+    - `분류`를 가진다.
+    - `탄산 여부`를 가진다.
+    - `칼로리`를 가진다.
```
>>>>칼로리는 원래 내용에 없었던 것 같은데, 회의를 통해서 여쭤보기

munkm1109 commented on base-cocktail-SRS.md:
```diff
@@ -0,0 +1,66 @@
+    - `칵테일 이름`으로 검색할 수 있다.
+    - `재료`로 검색할 수 있다.
+        - 여러 `재료`를 조합하여 검색할 수 있다.
+        - `알코올 유무`와 함께 검색할 수 있다.
```
앞서서 레시피 등록 부분에, "알코올 유무"기능이 없었습니다. '알코올 유무' 기능이 구현됐다면, 위에도 추가해야 할 것 같습니다.

munkm1109 commented on base-cocktail-SRS.md:
```diff
@@ -0,0 +1,66 @@
+    - `재료`로 검색할 수 있다.
+        - 여러 `재료`를 조합하여 검색할 수 있다.
+        - `알코올 유무`와 함께 검색할 수 있다.
+    - `알코올 유무`로만 검색할 수 있다.
```
알코올 유무에 대한 정의가 필요할 것 같습니다.
"알코올 유무로만 검색할 수 있다"는 말의 뜻이, 알코올이 있느냐 없느냐 라는 뜻이라면,  이 뜻도 이해할 수 있게 작성하면 좋을 것 같습니다.

munkm1109 commented on base-cocktail-SRS.md:
```diff
@@ -0,0 +1,66 @@
+- `검색어`로 `칵테일`을 검색할 수 있다.
+    - `칵테일 이름`으로 검색할 수 있다.
+    - `재료`로 검색할 수 있다.
+        - 여러 `재료`를 조합하여 검색할 수 있다.
```
재료를 조합하여 검색하는 방법의 구체적인 예시가 필요할 것 같습니다.
Ex) #설탕 #레몬
Ex2) 설탕, 레몬 ...

munkm1109 commented on base-cocktail-SRS.md:
```diff
@@ -0,0 +1,66 @@
+    - `기본 칵테일`만 볼 수 있다.
+    - `커스텀 칵테일`만 볼 수 있다.
+
+### 상세 정보
```
>>>>>디자인에, 페이지 상단 보면, 페이지 별 이름이 적혀있는데, 그 페이지의 이름을 따서 단락을 나누는거 어떻게 생각하시는 지 여쭤보기

munkm1109 commented on base-cocktail-SRS.md:
```diff
@@ -0,0 +1,66 @@
+    - `커스텀 칵테일`만 볼 수 있다.
+
+### 상세 정보
+- `칵테일 정보`를 조회할 수 있다.
```
'칵테일 정보' 란 무엇인지 정의하거나, 
앞서서 정의한 '칵테일'이 '칵테일 정보'와 같은 의미라면, '칵테일'로 통합해도 좋을 것 같습니다.

YoungH66 committed:
📝 Docs : basecocktail 요구사항정의서 초안 및 용어사전 초안 수정

코맨트와 디자인 시안, 회의 내용등을 반영하여 내용이 수정되었습니다.
 - 전반적인 내용을 정렬했습니다.
 - 등록 기준이라는 조건을 추가했습니다.
 - 다소 혼동될 수 있을 용어를 구분했습니다.
 - 용어사전에 내용이 추가되었습니다.
 - 이미지 요구사항이 수정되었습니다.
 - 칵테일 목록의 표시 방식을 수정, 반영했습니다.
 - 재료 정보를 구분하여 수정했습니다.
 - 비정형 데이터에 대한 처리 방식을 구상했습니다.
 - 추가 논의 사항을 정리해두었습니다.

YoungH66 commented:
요구사항정의서 및 용어사전 초안의 수정되었습니다. 확인 부탁드립니다.

YoungH66 committed:
📝 Docs : basecocktail 요구사항정의서 초안 및 용어사전 수정

0304 회의의 피드백을 수용하여 기존 요구사항 정의서 및 용어사전을 수정함.
 - 요구사항 정의서와 용어사전을 분리했습니다.
   - 요구사항 정의서 : base-cocktail-SRS.md
   - 용어사전 : base-cocktail-dictionary.md
 - 요구사항 정의서의 구조를 변경했습니다.
 - 구조는 trendcocktail 요구사항 정의서 및 용어사전을 참고했습니다.
 - 내용은 이전 수정버전과 크게 달라진 점 없이 구조 위주로 수정하였기 때문에 내용에 대한 피드백 요청드립니다.

Ref : #48

munkm1109 commented on base-cocktail-SRS.md:
```diff
@@ -0,0 +1,181 @@
+
+### 1.3. 재료 등록 기준
+- [ ] 재료 정보는 반드시 다음 조건을 만족해야 한다.
+  - [ ] 최소 1개 이상의 재료를 포함해야 한다.
```
디자인에 따르면, 베이스주류 / 추가재료 /가니쉬 로 재료가 구성되어있습니다.
베이스주류, 추가재료, 가니쉬로 나누어 조건을 적용해야할 것 같습니다.
가니쉬는 (선택)사항이기 때문에, (선택)사항에 대한 조건 또한 추가해야합니다

munkm1109 replied:
>>전체회의때 가니쉬 뺄지 논의

munkm1109 commented on base-cocktail-SRS.md:
```diff
@@ -0,0 +1,181 @@
+    - [ ] 숙성 기간: 해당되는 경우
+    - [ ] 증류 방식: 해당되는 경우
+    - [ ] 술의 역사: 주류에 대한 역사적 정보
+  - [ ] 재료 등록 순서는 중요하며, 처음 두 개의 재료는 칵테일의 대표 재료로 간주된다.
```
>>>>프론트, 디자이너와 이 기능 토의중입니다. 확정되면 말씀드릴게요.

munkm1109 commented on base-cocktail-SRS.md:
```diff
@@ -0,0 +1,181 @@
+    - [ ] 커스텀 칵테일 등록 시 첫 번째와 두 번째 재료 입력 필드는 특별히 표시되어야 한다(하이라이트 또는 알림)
+
+### 1.4. 레시피 등록 기준
+- [ ] 칵테일 정보는 반드시 다음 내용을 포함해야 한다.
```
레시피 등록 기준이 앞서 서술된 내용과 범위가 중복되는 것 같습니다. 
제 생각으로는, 
1.을 칵테일 레시피 등록으로 바꾸고, 
칵테일 레시피 등록에 사용되는 기능들을 
1.1.
1.2.
1.3. ...
으로 작성되면, 디자인-기능명세서-요구사항정의서 가 서로 호환되어서 보기 편할 것 같습니다.
>>혹시 영훈님께서 생각하신 방식이 있다면, 코멘트나, 간단하게 회의를 통해서 말씀해주실 수 있을까요?

munkm1109 commented on base-cocktail-SRS.md:
```diff
@@ -0,0 +1,181 @@
+### 1.4. 레시피 등록 기준
+- [ ] 칵테일 정보는 반드시 다음 내용을 포함해야 한다.
+  - [ ] 칵테일 이름(한글명, 영문명)
+  - [ ] 알코올 도수(ABV, %): 알코올이 포함된 경우 0% 초과, 포함되지 않은 경우 0%
```
제가 위에서 말씀드린대로 진행한다면, 재료 등록의, 베이스주류 등록기준에 들어가면 될것같습니다.

munkm1109 commented on base-cocktail-SRS.md:
```diff
@@ -0,0 +1,181 @@
+  - [ ] 칵테일 이름(한글명, 영문명)
+  - [ ] 알코올 도수(ABV, %): 알코올이 포함된 경우 0% 초과, 포함되지 않은 경우 0%
+  - [ ] 칵테일 분류: 칵테일의 종류(예: 하이볼, 마티니 등)
+  - [ ] 칵테일 맛 특성: 달콤함, 신맛, 쓴맛, 짠맛, 매운맛, 부드러운 맛 등 하나 이상 선택
```
>>차후에, 저희가 사용할 맛의 명칭을 정확히 전부 기술하고, ~등의 표현을 지우면 될 것 같습니다.
맛의 명칭은 제가 조사해보겠습니다.

munkm1109 commented on base-cocktail-SRS.md:
```diff
@@ -0,0 +1,181 @@
+  - [ ] 잘못된 URL로 연결된다.
+
+## 9. 추가 논의 필요 사항
+- [ ] 한글명-영문명 자동 변환 구현 여부 및 방식
```
전체회의때 효승님 의견 참고해서 결정하면 될것같습니다.

munkm1109 commented on base-cocktail-SRS.md:
```diff
@@ -0,0 +1,181 @@
+
+## 9. 추가 논의 필요 사항
+- [ ] 한글명-영문명 자동 변환 구현 여부 및 방식
+- [ ] 필터링 기능 구현 방식 및 범위:
```
>>목,금중에 한번 회의를 가질까요?
회의를 통해서 아래 안건들 처리하면 될것같습니다.
제가 모르는 부분들에 대해서 질문을 드려야 이해가 가능한 것도 섞여있어서요.

munkm1109 commented on base-cocktail-SRS.md:
```diff
@@ -0,0 +1,181 @@
+  - [ ] 기본/커스텀 칵테일 필터
+  - [ ] 알코올 도수 범위 필터
+  - [ ] 맛 특성 기반 필터
+- [ ] 커스텀 칵테일 대표 재료 입력 방식의 구체적인 UI/UX
```
안건 디자인, 프론트와 토의중입니다.

munkm1109 commented on base-cocktail-SRS.md:
```diff
@@ -0,0 +1,181 @@
+  - [ ] 각 제조 단계는 순서와 상세 설명을 포함해야 한다.
+  - [ ] 제조 단계는 번호가 매겨진 단계별 형식으로 작성해야 한다.
+
+### 1.5. 알코올 도수 계산 기준
```
계산 기능의 경우,
"알코올 도수의 값은 항상 0이상이어야 한다."
"성분의 비율의 합은 100이어야 한다."
등 수학적으로 전제되어야 하는 사항이 추가되어야 합니다.

YoungH66 committed:
📝 Docs : basecocktail 요구사항정의서 초안 및 용어사전 수정

0307 회의 요구사항에 따라 수정함.

munkm1109 commented on base-cocktail-SRS.md:
```diff
@@ -0,0 +1,205 @@
+
+### 1.1. 칵테일 등록
+- [ ] 칵테일을 등록할 수 있다.
+- [ ] 칵테일 유형을 구분할 수 있다.
```
칵테일 유형을 아래와 같이 구분할 수 있다.
로, 칵테일 유형에 대한 의미를 명확히 하는게 좋을것같습니다.

munkm1109 commented on base-cocktail-SRS.md:
```diff
@@ -0,0 +1,205 @@
+  - [ ] 기본 칵테일은 최소 2개 이상의 이미지를 가져야 한다.
+  - [ ] 커스텀 칵테일은 다음 방식 중 하나로 이미지를 등록해야 한다:
+    - [ ] 사용자가 직접 1개 이상의 이미지를 업로드
+    - [ ] 시스템에서 제공하는 기본 일러스트 이미지 중 하나를 선택
```
시스템에서 제공하는 기본 일러스트 이미지 제공기능이 없는걸로 알고있습니다.

munkm1109 commented on base-cocktail-SRS.md:
```diff
@@ -0,0 +1,205 @@
+
+### 1.3. 칵테일 이름 등록 기준
+- [ ] 칵테일 이름은 반드시 다음 조건을 만족해야 한다.
+  - [ ] 한글명과 영문명 입력 (회의 필요: 커스텀 칵테일의 경우 한글명만으로도 등록 가능 여부)
```
기본 칵테일의 이름만, 영문명을 지원.
재료, 커스텀칵테일의 이름은 한글만을 지원.

+@ 기본 칵테일 이름의 한글표기 중, 구글검색을 통해 검색이 불가한 경우, 부득이하게 한글표기와 영문표기를 복수 지원한다. (표기방식은 디자이너와 협의후 추가합니다. 차후 확인용 코멘트입니다)

munkm1109 commented on base-cocktail-SRS.md:
```diff
@@ -0,0 +1,205 @@
+    - [ ] 가니쉬(선택사항): 장식용 재료
+  - [ ] 각 재료는 다음 정보를 포함해야 한다.
+    - [ ] 재료명(한글명, 영문명)
+    - [ ] 용량: 숫자와 단위로 구성
```
ml를 제외한 모든 단위는, 구글검색결과 기준으로, ml로 환산하여 표기됩니다.
1 Table 스푼 = 15ml
1 Tea 스푼 = 5ml
1 shot = 45ml
1 oz = 30ml
1 dash = 1ml
1 jigger = 45ml
1 whips = 768ml
1 cup = 240ml
1 pint = 480ml
1 qt = 950ml
1 gallon = 3800ml

munkm1109 commented on base-cocktail-SRS.md:
```diff
@@ -0,0 +1,205 @@
+    - [ ] 용량: 숫자와 단위로 구성
+      - [ ] 기본 단위는 ml을 사용
+      - [ ] 비정형 단위("가득", "조금", "한 잔" 등)는 제조 과정에서만 사용 가능
+    - [ ] 재료 유형: 'alcohol'(주류) 또는 'other'(기타)
```
재료는 다음 세 가지로 분류된다 와 중복되는 내용입니다. 삭제해도 될것같습니다.

munkm1109 commented on base-cocktail-SRS.md:
```diff
@@ -0,0 +1,205 @@
+      - [ ] 기본 단위는 ml을 사용
+      - [ ] 비정형 단위("가득", "조금", "한 잔" 등)는 제조 과정에서만 사용 가능
+    - [ ] 재료 유형: 'alcohol'(주류) 또는 'other'(기타)
+  - [ ] 알코올(알코올 함유 재료)은 다음과 같은 추가 정보를 포함한다.
```
위스키(알코올)페이지 부분은, 신웅님께서 작성해주실거라, 여기 하위내용들 전부 지워도 될것같습니다. 
40번~48번까지

munkm1109 commented on base-cocktail-SRS.md:
```diff
@@ -0,0 +1,205 @@
+
+### 3.1. 칵테일 목록 표시
+- [ ] 기본 칵테일과 커스텀 칵테일이 혼합되어 표시된다.
+- [ ] 기본 칵테일은 "ORIGINAL"로 표시되며, 커스텀 칵테일은 "CUSTOM"으로 표시된다.
```
기본칵테일, 커스텀칵테일 표시되는 기능 논의된 기능일까요? 디자인상으론 ALL필터로 볼 때 구별없이 보여서, 혹시 논의된게 아니라면 빼도 될것같습니다

munkm1109 commented on base-cocktail-SRS.md:
```diff
@@ -0,0 +1,205 @@
+  - [ ] 칵테일 이름(한글명 표시)
+  - [ ] 대표 이미지
+  - [ ] 사용자의 좋아요 여부를 나타내는 아이콘
+  - [ ] 커스텀 칵테일인 경우 별도 표시
```
이것도 위와 동일합니다.

munkm1109 commented on base-cocktail-SRS.md:
```diff
@@ -0,0 +1,205 @@
+  - [ ] 대표 재료(베이스 주류)의 한글명
+
+### 3.2. 정렬 및 페이지네이션
+- [ ] 다음 기준으로 칵테일 목록을 정렬할 수 있다.
```
내림차순으로 전환하는 방법에 대한 서술이 필요할 것 같습니다.
저는, 각 정렬방법을 한번 더 누르는걸로 생각중인데, 일요일 전체회의에서 다뤄보고 정하면 될것같습니다.

munkm1109 commented on base-cocktail-SRS.md:
```diff
@@ -0,0 +1,205 @@
+  - [ ] 설명: 칵테일의 소개 및 배경 정보
+  - [ ] 기본 칵테일의 경우 최소 2개 이상의 이미지(대표 이미지, 보조 이미지)
+
+### 4.2. 알코올 상세 정보
```
이 부분은 삭제해도 될것같습니다.

munkm1109 commented on base-cocktail-SRS.md:
```diff
@@ -0,0 +1,205 @@
+### 5.1. 찜하기 기능
+- [ ] 사용자는 칵테일을 찜할 수 있다.
+- [ ] 찜한 목록을 조회할 수 있다.
+- [ ] 사용자는 찜한 칵테일을 찜 목록에서 제거할 수 있다.
```
찜목록에서 제거하는 방법을 서술해야할 것 같습니다. (디자이너와 논의 필요)

munkm1109 commented on base-cocktail-SRS.md:
```diff
@@ -0,0 +1,205 @@
+
+## 6. 시스템 구현
+
+### 6.1. 데이터 저장
```
효승님 의견에 따르면, 이런 기술적인 서술은 따로 백업해뒀다가, 객체기반의 모델링에 사용하시면 된다고 합니다. 요구사항정의서에서는 삭제해주시면 될것같습니다.
요구사항정의서는 모든사람이 보고 이해할 수 있는 내용으로 구성되어야 하기 때문입니다.

munkm1109 commented on base-cocktail-SRS.md:
```diff
@@ -0,0 +1,205 @@
+## 7. 향후 구현 예정 기능
+- [ ] 리뷰 기능: 사용자가 칵테일에 대한 리뷰를 작성하고 평점을 부여할 수 있는 기능
+
+## 8. 오류 처리
```
오류처리도 마찬가지로 객체기반 모델링에서 사용해주시면 될것같습니다.

YoungH66 committed:
📝 Docs : basecocktail 요구사항정의서 초안 및 용어사전 수정

munkm1109 commented on base-cocktail-SRS.md:
```diff
@@ -0,0 +1,160 @@
+### 3.1. 검색 인터페이스
+- [ ] 검색 아이콘 클릭 시 검색창이 표시된다.
+- [ ] 검색창에 텍스트를 입력할 수 있다.
+- [ ] 검색 결과를 표시한다.
```
돋보기로 하단에 여러개 나오는 디자인.
저거 검색결과가 아니라, Contain함수 사용한 자동완성 기능입니다.
예를들어 "다" 를 검색하면, 
"다"가 포함된 단어들이 아래에 주루룩 뜨는형식입니다.
물론 사전식 표기방식 오름차순으로요.

munkm1109 commented on base-cocktail-SRS.md:
```diff
@@ -0,0 +1,160 @@
+
+### 3.2. 검색 기능
+- [ ] 다양한 검색 조건을 지원한다:
+  - [ ] 이름 검색: 칵테일 또는 재료 이름으로 검색
```
재료이름은 아래에 재료검색 항목이 있어서, 여기선 "칵테일 검색" : 칵테일 이름으로 검색
으로 표현해도 될것같습니다.

munkm1109 commented on base-cocktail-SRS.md:
```diff
@@ -0,0 +1,160 @@
+  - [ ] 이름 검색: 칵테일 또는 재료 이름으로 검색
+  - [ ] 재료 검색: 포함된 재료로 검색
+  - [ ] 알코올 유무 검색: 알코올 함유 여부로 검색
+- [ ] 검색 결과 항목 클릭 시 해당 상세 페이지로 이동한다.
```
해당 상세 페이지가 아닌, 해당 "칵테일 상세페이지" 로 적어주셔야 합니다.

munkm1109 commented on base-cocktail-SRS.md:
```diff
@@ -0,0 +1,160 @@
+## 4. 칵테일 찾기
+
+### 4.1. 칵테일 목록 표시
+- [ ] 칵테일 목록을 그리드 형태로 표시한다.
```
그리드 형태가 무엇인지 용어사전에서 설명하거나,
그리드 형태 대신 누구나가 이해 가능한 알기쉬운 방식으로 표현하는게 좋습니다.

munkm1109 commented on base-cocktail-SRS.md:
```diff
@@ -0,0 +1,160 @@
+### 4.1. 칵테일 목록 표시
+- [ ] 칵테일 목록을 그리드 형태로 표시한다.
+  - [ ] 각 칵테일은 이미지, 이름, 좋아요 아이콘을 포함한다.
+  - [ ] 기본 칵테일의 이름은 영문 및 한글 표기를 지원하며, 일부 한글 표기가 어려운 경우 영문으로 표시할 수 있다.
```
기본칵테일은, 한글과 영문 병행표기를 하기로 됐습니다.
디자인에서 최근에 변경된 바나나스트로베리 크...을 참고하시면 됩니다.

munkm1109 commented on base-cocktail-SRS.md:
```diff
@@ -0,0 +1,160 @@
+  - [ ] 각 칵테일은 이미지, 이름, 좋아요 아이콘을 포함한다.
+  - [ ] 기본 칵테일의 이름은 영문 및 한글 표기를 지원하며, 일부 한글 표기가 어려운 경우 영문으로 표시할 수 있다.
+  - [ ] 커스텀 칵테일의 이름은 한글로 표시한다.
+  - [ ] 대표 재료(베이스 주류)를 한글로 표시한다.
```
영훈님이 말씀하셨던, 대표재료 2개가 선정되는 방식에 대해서도 서술하면 좋을 것 같습니다.
Ex) 대표재료 2개는, 칵테일 제조 페이지의 재료 등록 시, 최상단 2개 재료가 작성된다. 이런 내용을 매끄럽게 풀어쓰시면 될것같습니다.

munkm1109 replied:
++모든 재료는 한글로 표시된다는 내용도 추가하면 될것같습니다.

munkm1109 commented on base-cocktail-SRS.md:
```diff
@@ -0,0 +1,160 @@
+  - [ ] 이름순: 가나다 순으로 정렬
+  - [ ] 좋아요 수: 좋아요 많은 순으로 정렬
+  - [ ] 도수: 도수 높은/낮은 순으로 정렬
+  - [ ] 각 정렬 옵션을 한 번 더 클릭하면 오름차순/내림차순이 전환된다.
```
이건 잠깐 뺴두면 좋을것같습니다. 디자인-프론트랑 협의되지 않은 내용이기도 하고, 유저 편의성 측면에서, 핸드폰 화면에서 저 작은걸 한번 더누르는게 쉽지는 않아서요.

munkm1109 commented on base-cocktail-SRS.md:
```diff
@@ -0,0 +1,160 @@
+
+### 5.1. 칵테일 이미지
+- [ ] 칵테일의 대표 이미지를 표시한다.
+- [ ] 기본 칵테일의 경우 여러 이미지를 스와이프하여 볼 수 있다.
```
이건 디자인에 없는 내용이라서, 우선 빼야할것같습니다.

munkm1109 commented on base-cocktail-SRS.md:
```diff
@@ -0,0 +1,160 @@
+- [ ] 기본 칵테일의 경우 여러 이미지를 스와이프하여 볼 수 있다.
+
+### 5.2. 상단 기능 버튼
+- [ ] 찜하기 버튼: 칵테일을 찜 목록에 추가/제거할 수 있다.
```
찜하기버튼에 새로운 디자인이 추가되었는데, 
찜하기 제거 디자인이 추가되었습니다.
이걸 바탕으로 추가와 제거를 항목을 하나씩 분리해서 각각 서술해주시면 될것같습니다.

munkm1109 commented on base-cocktail-SRS.md:
```diff
@@ -0,0 +1,160 @@
+- [ ] 공유 버튼: 칵테일 정보를 다른 사람과 공유할 수 있다.
+
+### 5.3. 탭 메뉴
+- [ ] 정보, 재료, 제조법 탭을 제공하여 각 섹션으로 이동할 수 있다.
```
정보 = INFORMATION 처럼, 각각 표기된 이름이 다르기때문에, 정보는 어디로간다
재료는 어디로간다.
등 서술해주셔야 할 것 같습니다.

munkm1109 commented on base-cocktail-SRS.md:
```diff
@@ -0,0 +1,160 @@
+  - [ ] 재료명(한글)을 표시한다.
+  - [ ] 용량을 ml 단위로 표시한다. 다른 단위는 ml로 환산하여 표시한다.
+    - [ ] 1 Table 스푼 = 15ml, 1 Tea 스푼 = 5ml, 1 shot = 45ml, 1 oz = 30ml, 1 dash = 1ml 등으로 환산
+  - [ ] 비정형 단위("가득", "조금" 등)가 있는 경우 그대로 표시한다.
```
이부분은 저희가 회의한 내용 적용하면 될것같습니다.
가니쉬의 경우, OOO 가득 OOO1개 등의 비정형 단위를 이름 자체로 적용하여 표기한다. 이 내용입니다.

munkm1109 commented on base-cocktail-SRS.md:
```diff
@@ -0,0 +1,160 @@
+  - [ ] 각 단계는 번호와 설명으로 구성된다.
+  - [ ] 설명은 명확하고 간결하게 작성한다 (예: "잔에 얼음을 가득 채운다").
+
+## 6. 위스키 찾기 페이지
```
이부분은 여산님께서 작성해주시기때문에, 여산님께 참고하시면 된다고 말씀드리겠습니다.

YoungH66 committed:
📝 Docs : basecocktail 요구사항정의서 초안 및 용어사전 수정

YoungH66 committed:
📝 Docs : basecocktail 요구사항정의서 초안 및 용어사전 수정

YoungH66 committed:
📝 Docs : basecocktail 요구사항정의서 초안 및 용어사전 수정

YoungH66 committed:
📝 Docs : basecocktail 요구사항정의서 초안 및 용어사전 수정
